mathjax3-ts/output/chtml/FontData.ts

```diff
@@ -0,0 +1,328 @@
+        delim2:          1.0,
+        axis_height:      .25,
+        rule_thickness:   .06,
+        big_op_spacing1:  .111,
+        big_op_spacing2:  .167,
+        big_op_spacing3:  .2,
+        big_op_spacing4:  .45, // .6,  // better spacing for under arrows and braces
+        big_op_spacing5:  .1,
+
+        surd_height:      .075,
+
+        scriptspace:         .05,
+        nulldelimiterspace:  .12,
+        delimiterfactor:     901,
+        delimitershortfall:   .3,
+
+        min_rule_thickness:  1.25     // in pixels
+    };
+
+    /*
+     * The default delimiter and character data
+     */
+    protected static defaultDelimiters: DelimiterMap = {};
+    protected static defaultChars: CharMapMap = {};
+
+    /*
+     * The default variants for the fixed size stretchy delimiters
+     */
+    protected static defaultSizeVariants: string[] = [];
+
+    /*
+     * The actual variant, delimiter, and size information for this font
+     */
+    protected variant: VariantMap = {};
+    protected delimiters: DelimiterMap = {};
+    protected sizeVariants: string[];
+
+    /*
+     * The actual font parameters for this font
+     */
+    public params: FontParameters;
+
+    /*
+     * Copies the data from the defaults to the instance
+     *
+     * @constructor
+     */
+    constructor() {
+        let CLASS = (this.constructor as typeof FontData);
+        this.params = Object.assign({}, CLASS.defaultParams);
+        this.sizeVariants = CLASS.defaultSizeVariants.slice(0);
+        this.createVariants(CLASS.defaultVariants);
+        this.defineDelimiters(CLASS.defaultDelimiters);
+        for (const name of Object.keys(CLASS.defaultChars)) {
+            this.defineChars(name, CLASS.defaultChars[name]);
+        }
+    }
+
+    /*
+     * Creates the data structure for a variant (an object with prototype chain
+     *   that includes a copy of the linked variant, and then the inherit variant chain.
+     *   The linked copy is updated automatically when the link variant is modified.
+     *   (The idea is to be able to have something like bold-italic inherit from both
+     *   bold and intalic by having the prototype chain include a copy of bold plus
+     *   the full italic chain.  So if something is not defined explicitly in bold-italic,
+     *   it defaults first to a bold version, than an italic version, then the normal
+     *   version, which is in the italic prototype chain.)
+     *
+     * @param{string} name     The new variant to create
+     * @param{string} inherit  The variant to use if a character is not in this one
+     * @param{string} link     A variant to search before the inherit one (but only
+     *                           its top-level object).
+     */
+    public createVariant(name: string, inherit: string = null, link: string = null) {
+        let variant = {
+            linked: [] as CharMap[],
+            chars: (inherit ? Object.create(this.variant[inherit].chars) : {}) as CharMap
```

Comment:
This is not a clone, it creates an object having a prototype of the given object.  This is a lookup chain.  This allows a variant to have its own data (for for example for bold) but fall back on another variant (e.g., normal).  That way, the bold variant doesn't have to repeat all the data in the normal variant.

It is important to have inheritance of variants in this way, since, for example, one can do

```
<mstyle mathvariant="fraktur">
  <mi>x</mi>
  <mo>&gt;</mo>
  <mi>y</mi>
</mstyle>
```
so all three inner elements get `mathvariant="fraktur'`.  But there is no glyph for a plus sign in the MathJax Fraktur font, so it should default back to the normal version.  The prototype chain is what makes that work.